src/main/java/com/google/sps/data/TextAnalyser.java

```diff
@@ -0,0 +1,191 @@
+    try (LanguageServiceClient languageService = LanguageServiceClient.create(getSettings())) {
+      Document doc =
+        Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+      Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+        
+      return sentiment.getScore();
+    }
+  }
+
+  public String getMood() throws IOException {
+    float score = getSentimentScore();
+
+    if (Math.abs(score - 1) < EPSILON) {
+      return "very happy";
+    }
+ 
+    if (Math.abs(score + 1) < EPSILON) {
+      return "so pessimistic";
+    }
+ 
+    int position = 0;
+    String[] moods = new String[]{"neutral", "calm", "relaxed", "serene", "contented",
+                                  "joyful", "happy", "delighted", "excited", "thrilled",
+                                  "tense", "nervous", "stressed", "upset", "sad", 
+                                  "depressed", "bored", "fatigued", "pessimisctic"};
+ 
+    score = score * 10;
+    position = (int) score;
+ 
+    if (position < 0) {
+      position = position * (-1) + 9;
+    }
+ 
+    return moods[position];
+  }
+
+  private ClassifyTextResponse classify() throws IOException {
+    try (LanguageServiceClient language = LanguageServiceClient.create(getSettings())) {
+      Document document = 
+            Document.newBuilder().setContent(message).setType(Type.PLAIN_TEXT).build();
+      ClassifyTextRequest request = ClassifyTextRequest.newBuilder().setDocument(document).build();
+      
+      return language.classifyText(request);
+    } catch (Exception e) {
+      System.err.println("Not enough tokens (words) to actually get a category.");
+      return null;
+    }
+  }
+
+  // list of all the categories that text is about
+  // check https://cloud.google.com/natural-language/docs/categories
+  public Set<String> getCategories() throws IOException {
+    ClassifyTextResponse response = classify();
+
+    // no categories could be identified
+    if (response == null) {
+      return Collections.emptySet();
+    }
+      
+    Set<String> categories = new HashSet<String>();
+
+    for (ClassificationCategory category : response.getCategoriesList()) {
+      String[] listCategories = category.getName().split("/");
+      for (int i = 0; i < listCategories.length; i++) {
+        categories.add(listCategories[i].toLowerCase());
+        System.out.println(listCategories[i].toLowerCase());
+      }
+    }
+
+    return categories;
+  }
+
+  public Set<String> getEvents() {
+    Set<String> events = new HashSet<String>();
+    String[] allEvents = new String[]{"birthday", "wedding", "baby shower", "love",
+                                    "congratulation", "travel", "good morning",
+                                    "gratitude", "job", "promotion",
+                                    "new", "welcome", "good evening", "good night",
+                                    "holiday"};
+    String copy = message;
```

Comment:
No reason to have a new variable here, also Java doesn't really create copies (on memory) since Strings are immutable. You only create a second reference to the same String in memory.